### PR TITLE
DP-221 - Organisation Me Endpoint

### PR DESCRIPTION
**Pull request by maciej-goaco**

### CODE REVIEW

File: Services/CO.CDP.Organisation.WebApi/Model/Exceptions.cs [hunk 1]
```diff
@@ -9,4 +9,6 @@ public class InvalidUpdateSupplierInformationCommand(string message, Exception?
 public class InvalidQueryException(string message, Exception? cause = null) : Exception(message, cause);
 public class InvalidUpdateConnectedEntityCommand(string message, Exception? cause = null) : Exception(message, cause);
 
-public class UnknownConnectedEntityException(string message, Exception? cause = null) : Exception(message, cause);
\ No newline at end of file
+public class UnknownConnectedEntityException(string message, Exception? cause = null) : Exception(message, cause);
+
+public class MissingOrganisationIdException(string message, Exception? cause = null) : Exception(message, cause);
```
Comment (jakzal) [inline Services/CO.CDP.Organisation.WebApi/Model/Exceptions.cs:14]: This should result in a 404, so needs to be mapped to an error code. Like here https://github.com/cabinetoffice/GCGS-Central-Digital-Platform/blob/3a84adb9f963de84cc8a51ffcb5b4e9f2e2f6593/Services/CO.CDP.Organisation.WebApi/Extensions/ServiceCollectionExtensions.cs#L11

File: Services/CO.CDP.Organisation.WebApi/UseCase/GetOrganisationUseCase.cs [hunk 2]
```diff
@@ -1,12 +1,31 @@
...
 public class GetOrganisationUseCase(IOrganisationRepository organisationRepository, IMapper mapper) : IUseCase<Guid, Model.Organisation?>
 {
     public async Task<Model.Organisation?> Execute(Guid organisationId)
     {
+        return await organisationRepository.Find(organisationId)
+            .AndThen(mapper.Map<Model.Organisation>);
+    }
+}
+
+public class GetMyOrganisationUseCase(IOrganisationRepository organisationRepository, IMapper mapper, IClaimService claimService)
```
Comment (jakzal) [inline Services/CO.CDP.Organisation.WebApi/UseCase/GetOrganisationUseCase.cs:17]: Should be defined in its own file as it’s not related to the other use case.

File: Services/CO.CDP.Organisation.WebApi/Model/Exceptions.cs (hunk 1 above)
Comment (maciej-goaco) [reply Services/CO.CDP.Organisation.WebApi/Model/Exceptions.cs:14]: done.

File: Libraries/CO.CDP.Authentication/ClaimService.cs [hunk 3]
```diff
@@ -7,4 +7,9 @@ public class ClaimService(IHttpContextAccessor httpContextAccessor) : IClaimServ
     {
         return httpContextAccessor.HttpContext?.User?.FindFirst("sub")?.Value;
     }
+
+    public string? GetOrganisationId()
+    {
+        return httpContextAccessor.HttpContext?.User?.FindFirst("org")?.Value;
+    }
```
[the remark this answers is not in the document]
Comment (maciej-goaco) [reply Libraries/CO.CDP.Authentication/ClaimService.cs:18]: done.

**Review by jakzal — APPROVED** [review]

File: Services/CO.CDP.OrganisationInformation.Persistence/DatabaseOrganisationRepository.cs [hunk 4]
```diff
@@ -17,6 +17,14 @@ public void Dispose()
             .FirstOrDefaultAsync(t => t.Guid == organisationId);
     }
 
+    public async Task<Organisation?> Find(int organisationId)
```
Comment (glgoaco) [inline Services/CO.CDP.OrganisationInformation.Persistence/DatabaseOrganisationRepository.cs:20]: I recommend adding test coverage for the Find.

**Review by glgoaco — APPROVED** [review]

**Review by dharmverma — APPROVED** [review]

File: Services/CO.CDP.OrganisationInformation.Persistence/DatabaseOrganisationRepository.cs (hunk 4 above)
Comment (maciej-goaco) [reply Services/CO.CDP.OrganisationInformation.Persistence/DatabaseOrganisationRepository.cs:20]: good spot, added.

File: Services/CO.CDP.Organisation.WebApi/Extensions/ServiceCollectionExtensions.cs [hunk 5]
```diff
@@ -9,6 +9,7 @@ public static class ServiceCollectionExtensions
     private static readonly Dictionary<Type, (int, string)> ExceptionMap = new()
     {
         { typeof(BadHttpRequestException), (StatusCodes.Status422UnprocessableEntity, "UNPROCESSABLE_ENTITY") },
+        { typeof(MissingOrganisationIdException), (StatusCodes.Status422UnprocessableEntity, "UNPROCESSABLE_ENTITY") },
```
Comment (jakzal) [inline Services/CO.CDP.Organisation.WebApi/Extensions/ServiceCollectionExtensions.cs:12]: It should be 404. I considered 403 too, but we'd be giving out information to the caller this way. So 404.

The code should be something dedicated to this error code, i.e. `ORGANISATION_NOT_FOUND`.

**Review by jakzal — CHANGES_REQUESTED** [review]

File: Services/CO.CDP.Organisation.WebApi/Extensions/ServiceCollectionExtensions.cs (hunk 5 above)
Comment (maciej-goaco) [reply Services/CO.CDP.Organisation.WebApi/Extensions/ServiceCollectionExtensions.cs]: done.

**Review by jakzal — APPROVED** [review]

**Pull request merged**